base/compiler/abstractinterpretation.jl

```diff
@@ -932,13 +932,21 @@ function abstract_call_method_with_const_args(interp::AbstractInterpreter,
             return nothing
         end
         frame = InferenceState(inf_result, #=cache=#:local, interp)
-        frame === nothing && return nothing # this is probably a bad generated function (unsound), but just ignore it
+        if frame === nothing
+            add_remark!(interp, sv, "[constprop] Could not retrieve the source")
+            return nothing # this is probably a bad generated function (unsound), but just ignore it
+        end
         frame.parent = sv
-        typeinf(interp, frame) || return nothing
+        if !typeinf(interp, frame)
+            add_remark!(interp, sv, "[constprop] Constant inference failed")
```

Comment:
This seems a bit strong verbiage. It just hit a cycle, identical to the subsequent case (which differs because it found from the cache that the last attempt hit a cycle, instead of hitting it directly)
```suggestion
            add_remark!(interp, sv, "[constprop] Constant inference hit a cycle")
```

Reply:
Right, followed up at #47321.